Unable to build on Debian with`'clang: error: the clang compiler does not support '-march=native'`

## Context

I'm trying to compile Odin itself from source.

* Hardware: Raspberry Pi 4 (arm64)
* Operating System: Debian 12 (bookworm)
* Git tag: dev-2024-10

clang++ -v output:
```
Debian clang version 14.0.6
Target: aarch64-unknown-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
Found candidate GCC installation: /usr/bin/../lib/gcc/aarch64-linux-gnu/12
Selected GCC installation: /usr/bin/../lib/gcc/aarch64-linux-gnu/12
Candidate multilib: .;@m64
Selected multilib: .;@m64
```

## Expected Behavior

Compile Odin from source.

## Current Behavior

Clang is erroring out due to a incompatible flag.

## Failure Information (for bugs)

The linux distribution is up to date (as of time of writing) by running `apt-get update && apt-get upgrade`.

### Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1. git clone https://github.com/odin-lang/Odin.git
2. git checkout dev-2024-10
3. make release-native

### Failure Logs

```
./build_odin.sh release-native
+ clang++ src/main.cpp src/libtommath.cpp -Wno-switch -Wno-macro-redefined -Wno-unused-value -DODIN_VERSION_RAW="dev-2024-10" -DGIT_SHA="af9ae4897" -std=c++14 -I/usr/lib/llvm-14/include -std=c++14 -fno-exceptions -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -L/usr/lib/llvm-14/lib -O3 -march=native -pthread -lm -lstdc++ -ldl /usr/lib/llvm-14/lib/libLLVM-14.so -Wl,-rpath=$ORIGIN -o odin
clang: error: the clang compiler does not support '-march=native'
clang: error: the clang compiler does not support '-march=native'
make: *** [Makefile:19: release-native] Error 1
```

--- a/build_odin.sh
+++ b/build_odin.sh
@@ -130,7 +130,7 @@ build_odin() {
 		EXTRAFLAGS="-O3"
 		;;
 	release-native)
-		if [ "$OS_ARCH" = "arm64" ]; then
+		if [ "$OS_ARCH" = "arm64" ] || [ "$OS_ARCH" = "aarch64" ]; then
 			# Use preferred flag for Arm (ie arm64 / aarch64 / etc)
 			EXTRAFLAGS="-O3 -mcpu=native"
 		else

